src/Gremlin.Net.CosmosDb/Serialization/TreeJsonConverter.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Gremlin.Net.CosmosDb.Serialization.Trees;
+using Gremlin.Net.CosmosDb.Structure;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Gremlin.Net.CosmosDb.Serialization
+{
+    public class TreeJsonConverter : JsonConverter
+    {
+        public override bool CanWrite => false;
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer) { }
+
+        private static Dictionary<Type, Func<JsonReader, JsonSerializer, object>> _deserializers;
+        public TreeJsonConverter()
```

Comment:
This should be `static TreeJsonConverter()`. Otherwise, you'll instantiate the `_deserializers` field every time you new up a `TreeJsonConverter`.

Reply:
You could also just define the field as such and get rid of the constructor altogether:

```c#
private static Dictionary<Type, Func<JsonReader, JsonSerializer, object>> _deserializers = new Dictionary<Type, Func<JsonReader, JsonSerializer, object>>
{
    { typeof(Tree), ReadTree },
    { typeof(TreeVertexNode), ReadTreeVertexNode },
    { typeof(TreeEdgeNode), ReadTreeEdgeNode }
};
```